src/lsm/binary_search_benchmark.zig

```diff
@@ -142,24 +142,49 @@ const BenchmarkResult = struct {
 
 const Benchmark = struct {
     timer: std.time.Timer,
-    rusage: std.os.rusage,
+    utime_ns: u128,
 
     fn begin() !Benchmark {
         const timer = try std.time.Timer.start();
         return Benchmark{
             .timer = timer,
-            // TODO pass std.os.linux.rusage.SELF once Zig is upgraded
-            .rusage = std.os.getrusage(0),
+            .utime_ns = utime_nanos(),
         };
     }
 
     fn end(self: *Benchmark, samples: usize) !BenchmarkResult {
-        const rusage = std.os.getrusage(0);
+        const utime_now = utime_nanos();
         return BenchmarkResult{
             .wall_time = self.timer.read() / samples,
-            .utime = (timeval_to_ns(rusage.utime) - timeval_to_ns(self.rusage.utime)) / samples,
+            .utime = @intCast((utime_now - self.utime_ns) / samples),
         };
     }
+
+    fn utime_nanos() u128 {
```

Comment:
When we touch this code next time, I think we could remove this whole rusage tracking thing in favor of timer, following the rationale similar to per removal. 

https://github.com/tigerbeetle/tigerbeetle/commit/9702791c6008f4f69829debf935b6ac3a7cd5422